src/Xamarin.Android.Build.Tasks/Tasks/SetMSBuildExtensionsPath.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	public sealed class SetMSBuildExtensionsPath : Task
```

Comment:
The name seems to suggest that it sets the path to some given path. Since, it's trying to compute it internally, maybe it should renamed to indicate what it is doing or setting the path to?